yok.ts

```diff
@@ -167,8 +167,29 @@ export class Yok implements IInjector {
 		return _.contains(allCommands, name);
 	}
 
-	private buildHierarchicalCommand(parentCommandName: string, childCommandName: string) {
-		return util.format("%s|%s", parentCommandName, childCommandName);
+	public buildHierarchicalCommand(parentCommandName: string, commandLineArguments: string[]): any {
+		var subCommandName: string;
+		var subCommands = this.hierarchicalCommands[parentCommandName];
+		var remainingArguments = commandLineArguments;
+		var foundSubCommand = false;
+		_.each(commandLineArguments, arg => {
+			subCommandName = subCommandName ? [subCommandName, arg].join("|") : arg;
+			remainingArguments = _.rest(remainingArguments);
+			if(_.any(subCommands,(sc) => sc === subCommandName)) {
+				foundSubCommand = true;
+				return false;
+			} else if(_.any(subCommands, sc => sc === "*" + subCommandName)) {
+				subCommandName = "*" + subCommandName;
+				foundSubCommand = true;
+				return false;
+			}
+		});
+
+		if(foundSubCommand) {
+			return { commandName: [parentCommandName, subCommandName].join("|"), remainingArguments: remainingArguments };
```

Comment:
We use this `[parentCommandName, subCommandName].join("|")` on many places so I suggest to extract it to separate function.